### PR TITLE
Add include all networks

### PR DESCRIPTION
**Pull request by mbirghan**

### CODE REVIEW

File: src/features/featureincludeallnetworks.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef FEATURE_INCLUDEALLNETWORKS_H
```
Comment (bakulf) [inline src/features/featureincludeallnetworks.h:5]: I would call it "FeatureIOSKillSwitch". `IncludeAllNetworks` is just the API that allows us to implement the kill-switch.

File: src/features/featureincludeallnetworks.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#define FEATURE_INCLUDEALLNETWORKS_H
+
+#include "models/feature.h"
+
+constexpr const char* FEATURE_INCLUDEALLNETWORKS = "includeAllNetworks";
+
+class FeatureIncludeAllNetworks : public Feature {
+ public:
+  FeatureIncludeAllNetworks()
+      : Feature(FEATURE_INCLUDEALLNETWORKS, "Include All Networks",
```
Comment (bakulf) [inline src/features/featureincludeallnetworks.h:15]: Also here: "IOS Kill-Switch"

File: src/features/featureincludeallnetworks.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                L18nStrings::Empty,  // LongDescr
+                "",                  // ImagePath
+                "",                  // IconPath
+                "",                  // link URL
+                "2.8",               // released
+                true                // Can be enabled in devmode
+        ) {}
+
+  bool checkSupportCallback() const override {
+#if defined(MVPN_IOS)
```
Comment (bakulf) [inline src/features/featureincludeallnetworks.h:28]: Let's include this feature only on iOS. And to enforce it, At the top of this file:

```
#ifndef MVPN_IOS
#error "This feature should be exposed only to IOS - or a better error message here...
#endif
```

And here you can simply say: `return true;`

File: src/platforms/ios/ioscontroller.mm [hunk 4]
```diff
@@ -12,6 +12,7 @@
 #include "mozillavpn.h"
 #include "server.h"
 #include "settingsholder.h"
+#include "features/featureincludeallnetworks.h"
```
Comment (bakulf) [inline src/platforms/ios/ioscontroller.mm:15]: alphabetic order.

File: src/src.pro [hunk 5]
```diff
@@ -246,6 +246,7 @@ HEADERS += \
         features/featureinappaccountcreate.h \
         features/featureinappauth.h \
         features/featureinapppurchase.h \
+        features/featureincludeallnetworks.h \
```
Comment (bakulf) [inline src/src.pro:249]: Add it only in the `ios` block.

Comment (bakulf) [reply src/src.pro:249]: Plus you need to create the feature in `featurelist.cpp` but only `#ifdef MVPN_IOS`

**Review by bakulf — CHANGES_REQUESTED** [review]

File: src/featurelist.cpp [hunk 6]
```diff
@@ -18,6 +18,7 @@
 #include "features/featureinappaccountcreate.h"
 #include "features/featureinappauth.h"
 #include "features/featureinapppurchase.h"
+#include "features/featureioskillswitch.h"
```
Comment (bakulf) [inline src/featurelist.cpp:21]: `#ifdef MVPN_IOS`

File: src/featurelist.cpp [hunk 7]
```diff
@@ -56,6 +57,7 @@ void FeatureList::initialize() {
   new FeatureInAppAccountCreate();
   new FeatureInAppAuth();
   new FeatureInAppPurchase();
+  new FeatureIosKillswitch();
```
Comment (bakulf) [inline src/featurelist.cpp:60]: here too

File: src/src.pro [hunk 8]
```diff
@@ -855,6 +855,7 @@ else:ios {
             tasks/purchase/taskpurchase.h
 
     OBJECTIVE_HEADERS += \
+            features/featureioskillswitch.h \
```
Comment (bakulf) [inline src/src.pro:858]: move to the "HEADERS" block. I don't think it changes too much, but it's C++ and not obj-C++

**Review by bakulf — CHANGES_REQUESTED** [review]

File: src/features/featureioskillswitch.h [hunk 9]
```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef FEATURE_IOS_KILLSWITCH_H
+#define FEATURE_IOS_KILLSWITCH_H
+
```
[the remark this answers is not in the document]
Comment (mbirghan) [reply src/features/featureioskillswitch.h:7]: Sorry, I did not see this before. Will add it now.

**Review by bakulf — APPROVED** [review]

**Pull request merged**